labview/msg_headers.c

```diff
@@ -0,0 +1,270 @@
+#include <rabbitmq-c/amqp.h>
+
+#include "extcode.h"
+#include "labview_types.h"
+
+/**
+ * This function parses a string representation of headers and populates an AMQP
+ * table with the parsed header entries. The headers in the string are expected
+ * to be in a specific format where key-value pairs are separated by '=' and
+ * multiple headers are separated by ';'.
+ *
+ * @note The `table` parameter should be a initialized `amqp_table_t` structure.
+ *       The `headerBuffer` should point to a null-terminated C string.
+ *       The function modifies the `table` structure in place with the parsed
+ * entries.
+ */
+void stringToHeaders(amqp_table_t* table, const uint8_t* headerBuffer,
+                     uint64_t headerBufferLen) {
+  amqp_table_entry_t* entries = NULL;
+  int numEntries = 0;
+  int finished = FALSE;
+  int index = 0;
+
+  while (finished == 0) {
+    // Find the delimiter ';'
+    char* delimiter = findInBuffer(headerBuffer, headerBufferLen, index, 0x3B);
+    if (delimiter == NULL) {
+      delimiter =
+          headerBuffer +
+          headerBufferLen;  // If no delimiter found, use the end of the string
+    }
+    if (delimiter == headerBuffer + headerBufferLen) {
```

Comment:
why this extra check? in the previous if, you could write finished = TRUE already

Reply:
it covers a situation when a last char in the buffer is a delimiter. Then `findInBuffer` will return not null, but the algorithm should be finished.